### PR TITLE
refactor(@core/home/quote): refactor quote

### PR DESCRIPTION
**Pull request by marobakr**
1. apply the Single Responsibility Principle
  1.1 change `mobileMedia` to `isHandsetPortrait` and `minSmallMedia` to `isSmall`
  1.2 create a new function `updateFlagsForMatches` property based on the provided matches boolean.
  1.3 create a new function `updateViewportFlags` to check if the result matches true. call 'updateFlagsForMatches' 
    to update flags based on the provided matches' boolean
  1.4 create a new function `resetFlags` to reset the `isHandsetPortrait` and `isSmall` to false  if the result is not matched

2. rewrite valid comments for helper class

if applied, it will add changes to the quote components to make the code more readable, and apply the Single Responsibility Principle

### CODE REVIEW

File: src/app/core/components/home/quote/quote.component.html [hunk 1]
```diff
@@ -1,36 +1,32 @@
 <section
   [ngClass]="{
-    'small-media': mobileMedia,
-    'min-small-media': minSmallMedia
+    'small-media': isHandsetPortrait,
+    'medium-media': isSmall
   }"
-  class="qoute mx-auto position-relative"
```
Comment (vcpc) [inline src/app/core/components/home/quote/quote.component.html:6]: it's "quote" not "qoute"

File: src/app/core/components/home/quote/quote.component.html [hunk 2]
```diff
@@ -1,36 +1,32 @@
...
+  class="qoute mx-auto"
```
Comment (vcpc) [inline src/app/core/components/home/quote/quote.component.html:6]: it's "quote" not "qoute"

File: src/app/core/components/home/quote/quote.component.scss [hunk 3]
```diff
@@ -20,13 +21,13 @@
 
   /**
     * Helper to change the qoute in small screens
-    * 1. Change Positioning
-    * 2. Change Box Model
-    * 3. Change Text Size
+    * 1. reset the width to be fit content in small media
+    * 2. adding padding by X axis to prevent the quote aligned the viewport
```
Comment (vcpc) [inline src/app/core/components/home/quote/quote.component.scss:25]: not understandable ??

File: src/app/core/components/home/quote/quote.component.scss [hunk 4]
```diff
@@ -40,20 +41,14 @@
     }
   }
   /**
-    * Helper to change the qoute in medium and large screen
-    * 1. Change Positioning
-    * 2. Change Box Model
-    * 3. Change Text Size
+    * Helper to change the qoute in medium and small screen
+    * 1. reset the width to be fit content in medium-media
+    * 2. increase the font size to smaller in medium media
```
Comment (vcpc) [inline src/app/core/components/home/quote/quote.component.scss:46]: what does `2.` refer to?

**Comment by marobakr** [comment]
1. rename the rule name  to quote 
1.1 at line 6
1.2 at line 9
1.3 at line 12
1.4 at line 22
1.5 at line 26

2. rewrite valid comments for helper class
2.1 at line 23
1.2 at line 44

if applied, it will add changes to the quote components to make the code more readable

**Comment by marobakr** [comment]
1. apply the Single Responsibility Principle

if applied, it will add changes to the quote components to make the code more readable, and apply the Single Responsibility Principle

**Pull request merged**